pkg/collector/adapters/config_to_ports.go

```diff
@@ -60,28 +61,34 @@ func ConfigToReceiverPorts(logger logr.Logger, config map[interface{}]interface{
 		return nil, ErrReceiversNotAMap
 	}
 
+	sortedNames := make([]string, 0, len(receivers))
+	for name := range receivers {
+		sortedNames = append(sortedNames, name.(string))
```

Comment:
Would it make sense to check the type just for safety here?

```suggestion
		v, ok := name.(string)
		if !ok {
			logger.Warn("receiver does not match type, want: %s, got: %T", "string", name)
			continue
		}
		sortedNames = append(sortedNames, v)
```